core/src/com/unciv/logic/map/TileMap.kt

```diff
@@ -215,6 +217,42 @@ class TileMap {
         return viewableTiles
     }
 
+    /** Strips all units from [TileMap]
+     * @return stripped clone of [TileMap]
+     */
+    fun stripAllUnits(): TileMap {
+        val toReturn = TileMap()
+        toReturn.tileList = this.tileList.apply { forEach {it.stripUnits()} }
+        toReturn.mapParameters = mapParameters
+        return toReturn
+    }
+
+    /** Strips all units and starting location from [TileMap] for specified [Player]
+     * Operation in place
+     * @param player units of player to be stripped off
+     */
+    fun stripPlayer(player: Player) {
+        tileList.forEach {
+            if (it.improvement == "StartingLocation " + player.chosenCiv) { it.improvement = null }
+            for (unit in it.getUnits()) if (unit.owner == player.chosenCiv) unit.removeFromTile()
+        }
+    }
+
+    /** Finds all units and starting location of [Player] and changes their [Nation]
+     * Operation in place
+     * @param player player whose all units will be changed
+     * @param newNation new nation to be set up
+     */
+    fun switchPlayersNation(player: Player, newNation: Nation) {
+        tileList.forEach {
+            if (it.improvement == "StartingLocation " + player.chosenCiv) { it.improvement = newNation.name }
```

Comment:
it.improvement = "StartingLocation " + newNation.name

Reply:
Yeah, that was a typo. Fixed.